### PR TITLE
bug-1881333: switch to ruff format

### PR DESCRIPTION
**Pull request by willkg**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,15 +1,18 @@
 [tool.ruff]
+line-length = 88
+
+# NOTE(willkg): Need to keep using Python 3.8 until socorro-siggen drops
+# support for it
+target-version = "py38"
+
+[tool.ruff.lint]
```
Comment (willkg) [inline pyproject.toml:8]: ruff adjusted configuration when it expanded into formatting, so this moves some options to their correct table.

File: requirements.in [hunk 2]
```diff
@@ -47,7 +46,7 @@ pytest-env==1.1.3
 python-decouple==3.8
 requests==2.31.0
 requests-mock==1.11.0
-ruff==0.1.3
+ruff==0.2.2
```
Comment (willkg) [inline requirements.in:49]: While switching to ruff format, I updated ruff to the latest version which is 0.2.2.

File: pyproject.toml [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
 # Enable pycodestyle (E), pyflakes (F), and bugbear (B) rules
 select = ["E", "F", "B"]
 
 # Ignore line length violations--Black handles those
```
Comment (relud) [inline pyproject.toml:12]:
```suggestion
# Ignore line length violations
```

**Review by relud — APPROVED** [review]
r+wc

File: pyproject.toml (hunk 3 above)
Comment (willkg) [reply pyproject.toml]: Thank you!

**Pull request merged**